### PR TITLE
Fix kill fail bug

### LINKED ISSUE #11509: os.kill(pid, signal.SIGTERM) has no effect using paddlepaddle/paddle:latest-dev
Some unit tests like `test_listen_and_serv_op.py`, run the test with `paddlepaddle/paddle:latest-dev` docker image, the call `os.kill(pid, signal.SIGTERM)` have no effect, while run `kill pid` in the shell can successfully terminate the process. When using manylinux docker images, this issue is gone.

More info: https://github.com/PaddlePaddle/Paddle/pull/11470

### PR DESCRIPTION
**Pull request by velconia**
This close #11509

### CODE REVIEW

**Comment by velconia** [comment]
@luotao1 Could you please take a look at this? 

The missing of ```pip install opencv-python``` in Dockerfile will cause the unittest of distribution related op fail, so I add this command back and ```apt-get install python-opencv``` still exit.

According to the [issue](https://github.com/PaddlePaddle/Paddle/issues/10298), how could I check that this pr works for u in the flower dataset?

File: Dockerfile [hunk 1]
```diff
@@ -76,7 +76,8 @@ RUN easy_install -U pip && \
     pip install sphinx-rtd-theme==0.1.9 recommonmark
 
 RUN pip install pre-commit 'ipython==5.3.0' && \
-    pip install 'ipykernel==4.6.0' 'jupyter==1.0.0'
+    pip install 'ipykernel==4.6.0' 'jupyter==1.0.0' && \
+    pip install opencv-python
```
Comment (luotao1) [inline Dockerfile:80]: pip install opencv-python，和apt-get install 并不冲突， @velconia 已经测过，可以正常加载flowers数据集。

File: paddle/scripts/paddle_build.sh [hunk 2]
```diff
@@ -22,7 +22,7 @@
 function print_usage() {
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh:22]: paddle_build.sh的gen_dockerfile函数里面需要pip install 么？不然latest等镜像没有这个包。

Comment (velconia) [reply paddle/scripts/paddle_build.sh:22]: gen_dockerfile 里添加了 pip install /*.whl, 我理解会把 paddle python 依赖的所有包都加上, opencv-python也是其中之一

Comment (velconia) [reply paddle/scripts/paddle_build.sh:22]: @typhoonzero could you please take a look at this?

File: Dockerfile (hunk 1 above)
Comment (typhoonzero) [reply Dockerfile:80]: 为什么`pip install opencv-python`会影响进程terminate呢？

Comment (velconia) [reply Dockerfile:80]: 可以参考这个[issue](https://github.com/PaddlePaddle/Paddle/issues/11509#issuecomment-399122748)的回答

File: paddle/scripts/paddle_build.sh (hunk 2 above)
Comment (velconia) [reply paddle/scripts/paddle_build.sh:22]: whl installs packages here, in Paddle/python/setup.py.in

```
if '${CMAKE_SYSTEM_PROCESSOR}' not in ['arm', 'armv7-a', 'aarch64']:
    setup_requires+=['opencv-python']
```

File: python/paddle/fluid/tests/unittests/CMakeLists.txt [hunk 3]
```diff
@@ -43,12 +43,12 @@ list(REMOVE_ITEM TEST_OPS test_warpctc_op)
...
-# TODO(wuyi): this test hungs on CI, will add it back later
-list(REMOVE_ITEM TEST_OPS test_listen_and_serv_op)
 foreach(TEST_OP ${TEST_OPS})
     py_test_modules(${TEST_OP} MODULES ${TEST_OP})
 endforeach(TEST_OP)
 py_test_modules(test_warpctc_op MODULES test_warpctc_op ENVS FLAGS_warpctc_dir=${WARPCTC_LIB_DIR} SERIAL)
 py_test_modules(test_dist_train MODULES test_dist_train SERIAL)
 py_test_modules(test_parallel_executor_crf MODULES test_parallel_executor_crf SERIAL)
 py_test_modules(test_parallel_executor_fetch_feed MODULES test_parallel_executor_fetch_feed SERIAL)
+# (DONE) tests that need to be done in fixed timeout
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/CMakeLists.txt]: Can remove this comment

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**